nautobot/core/management/commands/pre_migrate.py

```diff
@@ -0,0 +1,114 @@
+import argparse
+import collections
+import itertools
+import sys
+
+from django.db import models
+from django.core.exceptions import ValidationError
+from django.core.management.base import BaseCommand
+
+from nautobot.dcim.models import VirtualChassis
+from nautobot.extras.models import ConfigContext, ConfigContextSchema, ExportTemplate
+
+
+HELP_TEXT = """
+Performs pre-migration validation checks for Nautobot 2.0.
+
+If the Nautobot 1.5 instance cannot be upgraded, this command will exit uncleanly.
+"""
+
+
+def check_virtualchassis_uniqueness():
```

Comment:
I wonder if we can use `check_for_duplicates_with_natural_key_fields_in_migration` in `nautobot.core.utils.migrations.py` for these checks?

Reply:
I thought about using some existing migration utils, but since this is going into `develop` I would have to just forklift the code. This is helpful! I could just forklift that and then use it across the board here.

Reply:
My bad i didnt even see it is based from `develop` but yeah should be somewhat useful!